src/auth.js

```diff
@@ -8,7 +8,13 @@ import { namespace, secrets } from './config';
  * we need to divide Date.now() by 1000, which otherwise would return miliseconds.
  */
 const isTokenValid = async () => {
-  const accessTokenExpireTime = await SecureStore.getItemAsync('ACCESS_TOKEN_EXPIRE_TIME');
+  let accessTokenExpireTime = null;
+  try {
+    accessTokenExpireTime = await SecureStore.getItemAsync('ACCESS_TOKEN_EXPIRE_TIME');
+  } catch (e) {
+    // this only throws if they reinstall on android
```

Comment:
who is "they"? and how is our code effected on reinstalls on android? should the code have a condition for android devices if this only effects android? what happens on ios?

Reply:
On Android, after the application is reinstalled, the value we want from the `SecureStore` must be deleted to recreate it.
I haven't tried it for iOS. That's my fault. I will try and write about my experiences. Maybe an extra logic can be added for Android. But I don't think we will have a problem on the iOS side. it will not enter the catch blog.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/fc0c2bbe7a779cb79d601672e1f4baa4b5eef5a5